packages/eslint-plugin/src/rules/naming-convention.ts

```diff
@@ -0,0 +1,1215 @@
+        'public',
+        'static',
+        'abstract',
+      ]),
+      ...selectorSchema('enumMember', false),
+
+      ...selectorSchema('typeLike', false, ['abstract']),
+      ...selectorSchema('class', false, ['abstract']),
+      ...selectorSchema('interface', false),
+      ...selectorSchema('typeAlias', false),
+      ...selectorSchema('enum', false),
+      ...selectorSchema('typeParameter', false),
+    ],
+  },
+  additionalItems: false,
+};
+
+// #endregion Schema Config
+
+// This essentially mirrors eslint's camelcase rule
+// note that that rule ignores leading and trailing underscores and only checks those in the middle of a variable name
+const defaultCamelCaseAllTheThingsConfig: Options = [
+  {
+    selector: 'default',
+    format: ['camelCase'],
+    leadingUnderscore: 'allow',
+    trailingUnderscore: 'allow',
+  },
+
+  {
+    selector: 'variable',
+    format: ['camelCase', 'UPPER_CASE'],
+    leadingUnderscore: 'allow',
+    trailingUnderscore: 'allow',
+  },
+
+  {
+    selector: 'typeLike',
+    format: ['PascalCase'],
+  },
+];
+
+export default util.createRule<Options, MessageIds>({
+  name: 'naming-convention',
+  meta: {
+    docs: {
+      category: 'Variables',
+      description:
+        'Enforces naming conventions for everything across a codebase',
+      recommended: false,
+      // technically only requires type checkin if the user uses "type" modifiers
+      requiresTypeChecking: true,
+    },
+    type: 'suggestion',
+    messages: {
+      unexpectedUnderscore:
+        '{{type}} name {{name}} must not have a {{position}} underscore.',
+      missingUnderscore:
+        '{{type}} name {{name}} must have a {{position}} underscore',
+      missingAffix:
+        '{{type}} name {{name}} must have one of the following {{position}}es: {{affixes}}',
+      doesNotMatchFormat:
+        '{{type}} name {{name}} must match one of the following formats: {{formats}}',
+    },
+    schema: SCHEMA,
+  },
+  defaultOptions: defaultCamelCaseAllTheThingsConfig,
+  create(contextWithoutDefaults) {
+    const context: Context = contextWithoutDefaults.options
+      ? contextWithoutDefaults
+      : // only apply the defaults when the user provides no config
+        Object.setPrototypeOf(
+          {
+            options: defaultCamelCaseAllTheThingsConfig,
+          },
+          contextWithoutDefaults,
+        );
+
+    const validators = parseOptions(context);
+
+    function handleMember(
+      validator: ValidatiorFunction | null,
+      node:
+        | TSESTree.Property
+        | TSESTree.ClassProperty
+        | TSESTree.TSAbstractClassProperty
+        | TSESTree.TSPropertySignature
+        | TSESTree.MethodDefinition
+        | TSESTree.TSAbstractMethodDefinition
+        | TSESTree.TSMethodSignature,
+      modifiers: Set<Modifiers>,
+    ): void {
+      if (!validator) {
+        return;
+      }
+
+      const key = node.key;
+      /* istanbul ignore if */ if (!util.isLiteralOrIdentifier(key)) {
```

Comment:
isLiteralOrIdentifier got remove in #1349 from `packages/eslint-plugin/src/util/misc.ts`

```ts
/**
 * This covers both actual property names, as well as computed properties that are either
 * an identifier or a literal at the top level.
 */
function isLiteralOrIdentifier(
  node: TSESTree.Expression,
): node is TSESTree.Literal | TSESTree.Identifier {
  return (
    node.type === AST_NODE_TYPES.Literal ||
    node.type === AST_NODE_TYPES.Identifier
  );
}
```

----------------

do you mind removing `istanbul ignore` statements?